### ISSUE TITLE
Wedged polar plot's position/scaling leaves too much white space.

### ISSUE THREAD

**Issue by ImportanceOfBeingErnest** [matplotlib/matplotlib#10264]
Coming from [this stackoverflow question](https://stackoverflow.com/questions/48289129/removing-space-around-wedge-polar-plots-in-matplotlib), some strange (to me) behaviour is observed when plotting partial or wedged polar plots. This new feature, which can be used by setting `.set_thetamin` and `.set_thetamax` on a polar plot, allows to plot only part of a polar plot without the need to use [floating axes](https://matplotlib.org/2.0.2/mpl_toolkits/axes_grid/users/overview.html#floating-axes) - which is of course a great feature. 

The problem arises when either using those partial polar plots as subplots or when trying to plot them tightly in a figure. It seems like their position is always set to be in the middle of some kind of subplot with aspect ratio of 1. Especially when using a half circle polar plot, this causes a lot of unnecessary white space around the half-circle. 

Consider this code:

```python
import numpy as np
import matplotlib.pyplot as plt

fig, axes = plt.subplots(1, 2, subplot_kw={'projection': 'polar'},
                         figsize=(8, 4))

for i, theta_max in enumerate([2*np.pi, np.pi]):

    # define theta vector with varying end point and some data to plot
    theta = np.linspace(0, theta_max, 181)
    axes[i].set_thetamin(0)
    axes[i].set_thetamax(theta_max*180/np.pi)

    axes[i].set_xlabel('Magnitude', fontsize=15)
    axes[i].set_ylabel('Angles', fontsize=15)

plt.savefig("polarwedges2.svg", facecolor="#e8f4f2")
```

which produces the top plot in the below image. I have manually marked the region which is occupied by the polar plot, calling it "Incompressible BBox", because it occupies much more space than desired, also making the xlabel appearing totally off.  The picture below is what I would expect this subplot to behave like.

![image](https://user-images.githubusercontent.com/23121882/35073482-47063acc-fbe9-11e7-9034-6bdba4ec5331.png)

The "imcompressibility" of that box also prevents to have such a half-circle polar plot tight in a figure. 
In the following code a simple figure with such a plot is attempted.

```python
import matplotlib.pyplot as plt

fig, ax = plt.subplots(1, 1, figsize=(6,3.4), subplot_kw={'projection': 'polar'})

# set 'thetamin' and 'thetamax' according to data
ax.set_thetamin(0)
ax.set_thetamax(180)

ax.set_xlabel('Magnitude', fontsize=15)
ax.set_ylabel('Angles', fontsize=15)

plt.savefig("polarwedge3.svg", facecolor="#e8f4f2")
```
The plot appears much smaller than the figure would allow for and leaves a lot of white space around.

![image](https://user-images.githubusercontent.com/23121882/35074175-f5463698-fbec-11e7-9aaf-9111df0a363e.png)

The only workaround I can currently think of is to make the original axes much larger than the figure (`ax.set_position( [0.1, -0.45, 0.8, 2])`) and shift the xlabel manually by some amount towards the top, as shown in the lower picture. Workaround code.

```python
import matplotlib.pyplot as plt

fig, ax = plt.subplots(1, 1, figsize=(6,3.4), subplot_kw={'projection': 'polar'})

# set 'thetamin' and 'thetamax' according to data
ax.set_thetamin(0)
ax.set_thetamax(180)

ax.set_xlabel('Magnitude', fontsize=15, labelpad=-60)
ax.set_ylabel('Angles', fontsize=15)

ax.set_position( [0.1, -0.45, 0.8, 2])
plt.savefig("polarwedge4.svg", facecolor="#e8f4f2")
```

I guess this behaviour is in general undesired. One would rather have either the same behaviour as for fixed aspect plots (like imshow, i.e. what is called "compressible" above) or even better, some API to set the position to useful values according to the usual subplot mechanism.

**Comment by QuLogic** [comment]
This is expected (by me because I wrote it) behaviour, though not really desired. Polar plots set themselves to have equal aspect by default. This is only done in `__init__` though, so wedges are simply zoomed as much as possible to fit in that square box.

Since the space for the Axes is provided by the Figure, I never really knew how to get it to resize itself, and still don't really know. If anyone knows, I'd like to find out, because it's something that seems desirable in other places, like Cartopy for instance.

**Comment by WeatherGod** [comment]
This seems like something a constrained layout manager might be able to
handle? Maybe somewhat?

Allowing a plot to resize a figure is currently a big no-no in matplotlib.
It is technically possible, but it would be a really bad idea in practice.
What I do personally is to call matplotlib.figure.figaspect() to give me a
reasonable figure size ahead of time.

**Comment by QuLogic** [comment]
Well, it's not resizing the Figure that's needed, "just" resizing the space needed for the Axes. But I think that's still managed by the Figure itself, no?

**Comment by ImportanceOfBeingErnest** [comment]
I have to admit that my understanding of the internals is rather limited. But since normal image subplots are handled exactly as desired (i.e. their axes size is as large as the image itself by default) I would have thought that there is a mechanism that provides this behaviour and which can be used here as well.

**Comment by jklymak** [comment]
First there is a constrained layout manager PR #9082.  

> Since the space for the Axes is provided by the Figure, I never really knew how to get it to resize itself, and still don't really know. If anyone knows, I'd like to find out, because it's something that seems desirable in other places, like Cartopy for instance.

The space that the axes is being drawn is set by `subplots`, where it returns a `position` for the axes, called the "original" position.  If you set the x/ylim of this axis, and then set the `aspect` then the there is an "active" position that tells us where the new axis is.  <aside> Bizarrely, that new position doesn't calculated until draw time, which is not of too much use if you call `get_position(original=False)`.  (please see #9855 which fixes this, but hasn't been reviewed yet).  </aside>

I don't know the internals of the polar projection.  if I was just drawing half a circle on a Cartesian projection, I'd set the `ylim = [0.5 1.], xlim=[0., 1.]` and then the "active" axis would be the right aspect ratio.  

I'd need to play w/ the above, but it seems maybe polar projection should have some awareness of the extent it wants to draw over and modify its transform accordingly.  It may need to modify the "active" axis position as well, but I don't know how polar does that at the moment.

**Comment by QuLogic** [comment]
`xlim` and `ylim` are in data coordinates, so that wouldn't work (your `xlim` is 0 to 57 degrees). After looking around, I do think some fiddling with the "active" position could probably work, but it would require some tweaking of internals that assume square equal aspect ratio.

**Comment by jklymak** [comment]
Yeah, and I don't understand the Transforms.  Whats happening now is that `set_aspect('equal')` is defining a cartesian box in the middle of the axis with relative limits 0-1, 0-1 that I presume the transform now maps into.  

I *think* that aspect should be changed from 1 to 0.5 (in the present case, or whatever fraction of the circle is being made) and the y part of the transform multiplied by 2.0.  

If I `set_aspect` in PolarAxes, I get half way there with the x-axis stretched.  But I don't understand the Transforms well enough to then get the y-direction to stretch and be offset down.

**Comment by QuLogic** [comment]
Yes, I've found something similar. Partially, the problem is in [the wedge bbox here](https://github.com/matplotlib/matplotlib/blob/b25bb5c4b650f397c7c81f972b525367db7843ff/lib/matplotlib/projections/polar.py#L818) because it tries extra hard to get equal aspect ratio. I can set aspect ratio in `draw` where the wedge is updated, but even if that adjustment bitis ignored, there's still some other bit wrong as clipping is not quite right.

**Comment by karlos124c** [comment]
excellent

**Comment by VladimirIvonin** [comment]
Any updates?

**Comment by QuLogic** [comment]
I have made some progress; the two images from the original post now produce:
![Figure_1](https://github.com/user-attachments/assets/de391b5b-af74-4202-9f41-4352542063c7)
![Figure_2](https://github.com/user-attachments/assets/bd738ac5-aa02-444f-a2c4-decad706fb4f)

I need to clean up the code a bit, but it appears to work nicely in all cases. As you can see above though, the Axis labels might still need some tweaking, as I think they don't account for the optional radial ticks turning on automatically.

**Comment by juanPabloMiceli** [comment]
Any updates in merging this PR?
